### ISSUE TITLE
Configuration ignored when in a workspace subdirectory

### ISSUE THREAD

**Issue by gdlx** [microsoft/vscode-flake8#194]
I have a python project nested in a git repo, and I'm storing a `tox.ini` file (I tried with `setup.cfg` & ` .flake8`) at the root of the python project, but they are ignored by the extension.

```
workspace-root/ (/home/user/work/myproject)
├─ pythonproject/
│  ├─ module/
│  │  ├─ common.py
│  ├─ tox.ini
```

The cause is that the extension runs like this flake8 from the workspace root folder, as shown in the log:
```
2023-08-01 21:53:26.869 [info] CWD Linter: /home/user/work/myproject
2023-08-01 21:53:43.513 [info] [...]/python -m flake8 --format='%(row)d,%(col)d,%(code).1s,%(code)s:%(text)s' /home/user/work/myproject/pythonproject/module/common.py
```

Here is my `tox.ini`:
```ini
[flake8]
extend-ignore = E203
max-line-length = 120
```

When I'm using the same command from the CLI in the workspace root, the config is also ignored:
```
78,80,E,E501:line too long (94 > 79 characters)
94,80,E,E501:line too long (88 > 79 characters)
104,80,E,E501:line too long (84 > 79 characters)
131,80,E,E501:line too long (88 > 79 characters)
174,80,E,E501:line too long (85 > 79 characters)
211,80,E,E501:line too long (86 > 79 characters)
272,80,E,E501:line too long (88 > 79 characters)
296,25,E,E203:whitespace before ':'
```

When running it from the python subproject path (where `tox.ini` is), no error is displayed, the config just works as expected.

The problem is that flake8 seems to search config files from the current directory, and then in the parents, as explained here: https://github.com/PyCQA/flake8/issues/1111#issuecomment-812888257

Of course, the extension cannot guess where my config file is (well...or maybe it could ?), but I think it should execute the command from the checked file's directory and let flake8 find the closest config as expected, which I tested as also working fine:
```
cd /home/user/work/myproject/mypythonproject/mymodule
[python path]/bin/python -m flake8 --format='%(row)d,%(col)d,%(code).1s,%(code)s:%(text)s' common.py
```

Of course I could set the flake8 config file option in the workspace config, but each project contributors would have to do the same, which would be a bummer knowing flake8 has the ability to find the closest config (yet I wish it had an option to search the config from the tested file directory, nevermind the working dir...)

I also could move `tox.ini` in the workspace root, but it would also be ugly, and what if I have multiple python subprojects with different flake8 configs ?

**Comment by karthiknadig** [comment]
A workaround, while we consider this issue, is to use `"flake8.args" = ["--config=<file>"]`, you can use `${workspaceFolder}/pythonproject/tox.ini` as the path.

> what if I have multiple python subprojects with different flake8 configs ?

A way to handle multiple packages is to use the multi-root workspace feature in VS code.  See https://code.visualstudio.com/docs/editor/multi-root-workspaces . All you need is a `project.code-workspace` file, see example here: https://code.visualstudio.com/docs/editor/multi-root-workspaces#_workspace-file-schema

**Comment by gdlx** [comment]
Thanks !

I'm actually using the first workaround for now but it's quite static while flake8 is actually compatible with multiple projects in the same repo.

I'd use the 2d workaround whenever a second python project is added, but splitting a monorepo into multiple roots wouldn't be very handy.

**Comment by karthiknadig** [comment]
> splitting a monorepo into multiple roots wouldn't be very handy

Agreed, that is a temp solution. We will look setting `cwd` to file dir.
